src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -667,6 +668,62 @@ def create_application(client, display_name, homepage=None, identifier_uris=None
     return result
 
 
+def grant_application(cmd, identifier, appid, expires='1'):
+    if not appid:
+        raise CLIError("'--appid' is required for grant OAuth2 permissions")
+
+    graph_client = _graph_client_factory(cmd.cli_ctx)
+
+    # Get the Service principal ObjectId for the client app
+
+    sps = list(graph_client.service_principals.list(
+        filter="appId eq '{}'".format(identifier)))
+
+    if not sps:
+        raise CLIError("'--id'" + identifier + " does not exist on this directory")
+    client_spn = sps[0].object_id
+
+    # Get the Service principal ObjectId for associated AAD Apps
+
+    if appid == '00000002-0000-0000-c000-000000000000':  # This is AD Read
+        scope = 'User.Read'
+    else:
+        scope = "user_impersonation"
```

Comment:
It this a right assumption? If you like to focus on common permission, I suggest add a `elif` by comparing with the real id  of `impersonation` and then keep the guid for the rest. Also move the code block to end since this is only for display.

Reply:
We pass the scope into the payload and is required to grant permissions, so we can't move it, it is required. This is a common well know AAD GUID and as far as I know this scope needs to be set differently as per the code block.

Reply:
Want to get on the same page with you.  The `00000002-0000-0000-c000-000000000000`, to me, is the resource id of AAD graph. `Scope` in this context, means delegated permissions. Right? 
If yes, the code here means if the resource is AAD graph, the delegated permission will be `READ`. For native apps, this permission is way too limited, can we also use `user_impersonation`, or expose as an argument if we can't make a reasonable default here?

Reply:
Removed this for now and we can re-implement if required